doc/libpmemobj/pmemobj_ctl_get.3.md

```diff
@@ -234,14 +234,22 @@ This entry point takes a complex argument.
 
 ```
 struct pobj_alloc_class_desc {
+	size_t size;
 	size_t unit_size;
 	unsigned units_per_block;
 	enum pobj_header_type header_type;
 	unsigned class_id;
 };
 ```
 
-The first field, `unit_size`, is an 8-byte unsigned integer that defines the
+The `size` field is the size of the argument structure. This field should NOT
+be included when creating string queries. Normally, setting this field takes the
```

Comment:
"string queries" seems out of place here. I think it should be mentioned later where string queries are explained.

Also, skipping one field in string query is a bit strange... I see 2 other options:
- using version instead of size; user will have to specify version in string query, eg:
   heap.alloc_class.128.desc=v1,500,1000,compact
- adding support for named attributes in string queries, eg:
   heap.alloc_class.128.desc=unit_size=500,units_per_block=1000,header_type=compact

Reply:
I'm not going to oppose this much, but we have to be very careful. Adding size instead of version closes the possibility of changing layout of this structure.